sourcegraph/sourcegraph.proto

```diff
@@ -142,7 +146,9 @@ message Repo {
 	// Fork is whether this repository is a fork.
 	bool fork = 12;
 
-	// Mirror is whether this repository is a mirror.
+	// Mirror indicates whether this repo's canonical location is on
+	// another server. Mirror repos track their upstream and are not
+	// eligible for discovery on this server.
 	bool mirror = 13;
```

Comment:
Why would you ever want to mirror something hosted on another Sourcegraph instance? In that case, shouldn't you just fork that repository to your own local instance? The relationship of mirrored vs non-mirrored federated  vs non-mirrored non-federated is pretty complex and confusing. It'd be better to enforce a simple "right" way of doing things, even if it means some edge cases aren't covered IMO. If people really want to copy a repository, they can always fork it, right?